cloudapp/src/app/main/main.component.ts

```diff
@@ -127,6 +129,19 @@ export class MainComponent implements OnInit {
   }
 
 
+  onExcel() {
+    this.alert.clear()
+    this.saveOptions()
+
+    let libraryCode:string = this.form.value.libraryCode.trim()
+    let circDeskCode:string = this.form.value.circDeskCode.trim()
+    let columnOptions:ColumnOption[] = this.form.value.columnOptionsList.filter(c => c.include)
+    this.loading = true
+    this.excelExportService.generateExcel(circDeskCode, libraryCode, columnOptions)
```

Comment:
generateExcel is an async function. So I think you need to make onExcel an async function and await the result of generateExcel before setting this.loading back to false.

Reply:
Resolved with commit b075a3c